polars/polars-core/src/chunked_array/strings/encoding.rs

```diff
@@ -0,0 +1,53 @@
+use crate::prelude::*;
+
+use base64;
+use hex;
+
+impl Utf8Chunked {
+    #[cfg(feature = "string_encoding")]
+    pub fn hex_decode(&self, strict: Option<bool>) -> Result<Utf8Chunked> {
```

Comment:
Not a blocker for this PR, but I think we can reduce compilation time if we implement this as a kernel in polars-arrow. Then we dipatch this logic to that crate with `apply_kernel`

Reply:
i was trying to get it to work with apply_kernel, but was having some difficulties.  I couldnt figure out how to have the `decode` return a `Result<Option<T>>` return type. Likely because im still a bit unfamiliar with some of the lower level workings of the chunkedArrays like `apply_kernel`

Reply:
Hmm.. Yes, it seems like we should also add a `try_apply_kernel`. Let's follow this up in a future PR.